app/models/supervisor_volunteer.rb

```diff
@@ -5,10 +5,19 @@ class SupervisorVolunteer < ApplicationRecord
   belongs_to :supervisor, class_name: "User"
   validates :supervisor_id, uniqueness: {scope: :volunteer_id} # only 1 row allowed per supervisor-volunteer pair
   validates :volunteer_id, uniqueness: {scope: :is_active}, if: :is_active?
+  validate :ensure_supervisor_and_volunteer_belong_to_same_casa_org, if: -> { supervisor.present? && volunteer.present? }
 
   def is_active?
     is_active == true
   end
+
+  private
+
+  def ensure_supervisor_and_volunteer_belong_to_same_casa_org
```

Comment:
This is great!
We will need a lot of this in the codebase. I want to think through how to systematize it rather than having it in models here and there.

Reply:
Were you thinking something along these lines? https://github.com/rubyforgood/casa/compare/main...patrickarnett:validates-shared-attribute